src/main/java/com/statful/sender/MetricsHolder.java

```diff
@@ -64,26 +71,44 @@ public final boolean addMetric(final DataPoint dataPoint) {
         }
 
         boolean inserted = this.buffer.offer(dataPoint);
+
+        this.shouldFlush(inserted);
+
         if (!inserted) {
             LOGGER.warn("metric could not be added to buffer, discarding it {} ", dataPoint.toMetricLine());
         }
         return inserted;
     }
 
     /**
+     * If a metric could not be added to the buffer tries to flush the buffer.
+     * Also check if a the buffer has at least as many items as the flush size and tries to flush the buffer
+     */
+    private void shouldFlush(final boolean inserted) {
+        if (!inserted) {
+            this.flush();
+            return;
+        }
+
+        int size = this.buffer.size();
```

Comment:
We are not doing it. There is a return inside the if, but I'll move it to an else block to make it clearer.